Various build problems.

Had a few problems building.

The first is trivial, but might confuse beginners.  In the README.md
```git clone git@github.com:ComparativeGenomicsToolkit/hal.git```

Unless they have commit permission should be:
```git clone git://github.com/ComparativeGenomicsToolkit/hal.git```

On Ubuntu 18.04 there's a strange problem with the make file, where the makefile fails if the libs are not executable:
```
ar rc ../lib/libHal.a ../objs/api/impl/halAlignmentInstance.o ../objs/api/impl/halMappedSegment.o ../objs/api/impl/halColumnIterator.o ../objs/api/impl/halSegmentIterator.o ../objs/api/impl/halTopSegment.o ../objs/api/impl/halBottomSegmentIterator.o ../objs/api/impl/halSegmentMapper.o ../objs/api/impl/halValidate.o ../objs/api/impl/halGappedBottomSegmentIterator.o ../objs/api/impl/halTopSegmentIterator.o ../objs/api/impl/halCLParser.o ../objs/api/impl/halRearrangement.o ../objs/api/impl/halPositionCache.o ../objs/api/impl/halCommon.o ../objs/api/impl/halSegment.o ../objs/api/impl/halBottomSegment.o ../objs/api/impl/halGappedTopSegmentIterator.o ../objs/api/impl/halGenome.o ../objs/api/hdf5_impl/hdf5Sequence.o ../objs/api/hdf5_impl/hdf5DnaDriver.o ../objs/api/hdf5_impl/hdf5BottomSegment.o ../objs/api/hdf5_impl/hdf5SequenceIterator.o ../objs/api/hdf5_impl/hdf5Genome.o ../objs/api/hdf5_impl/hdf5TopSegment.o ../objs/api/hdf5_impl/hdf5ExternalArray.o ../objs/api/hdf5_impl/hdf5UDCFuseDriver.o ../objs/api/hdf5_impl/hdf5Alignment.o ../objs/api/hdf5_impl/hdf5MetaData.o ../objs/api/mmap_impl/mmapAlignment.o ../objs/api/mmap_impl/mmapGenomeSiteMap.o ../objs/api/mmap_impl/mmapDnaDriver.o ../objs/api/mmap_impl/mmapPerfectHashTable.o ../objs/api/mmap_impl/mmapRbTree.o ../objs/api/mmap_impl/mmapBottomSegment.o ../objs/api/mmap_impl/mmapSequence.o ../objs/api/mmap_impl/mmapFile.o ../objs/api/mmap_impl/mmapGenome.o ../objs/api/mmap_impl/mmapPhf.o ../objs/api/mmap_impl/mmapTopSegment.o 
../lib/libHal.a
make[1]: execvp: ../lib/libHal.a: Permission denied
../rules.mk:39: recipe for target '../lib/libHal.a' failed
make[1]: *** [../lib/libHal.a] Error 127
make[1]: Leaving directory '/share/apps/ComparativeGenomicsToolkit/hal/api'
Makefile:13: recipe for target 'api.libs' failed
make: *** [api.libs] Error 2
```
The ugly fix is:
```chmod +x ./lib/*.a; make```

But you have to do that for each lib so it takes 7 times.

--- a/README.md
+++ b/README.md
@@ -36,11 +36,15 @@ Installation
 
 From the parent directory of where you want HAL installed:
 
-	 git clone git@github.com:ComparativeGenomicsToolkit/hal.git
+	 git clone https://github.com/ComparativeGenomicsToolkit/hal.git
 
 ### Installing Dependencies
 
 #### HDF5 1.10.1 with C++ API enabled
+
+* Using apt (Ubuntu 18.04)
+
+    sudo apt install libhdf5-dev
 
 * Using [MacPorts](http://www.macports.org/):   
 
@@ -82,7 +86,7 @@ From the parent directory of where you want HAL installed:
 
 From the same parent directory where you downloaded HAL:
 
-	  git clone git://github.com/benedictpaten/sonLib.git
+	  git clone https://github.com/ComparativeGenomicsToolkit/sonLib.git
 	  pushd sonLib && make && popd
 
 If sonLib and HAL are not sister directories, update hal/include and change

